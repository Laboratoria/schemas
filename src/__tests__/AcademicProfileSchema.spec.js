const mongoose = require('mongoose/browser');
const {
  UserSchema,
  AcademicProfileSchema,
  AcademicProfileTagSchema,
  AcademicProfileEndorsementSchema,
  AcademicProfileCommentSchema,
} = require('../../')(mongoose);

describe('AcademicProfileSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, AcademicProfileSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', () => {
    const user = new mongoose.Document({}, UserSchema);
    const tag = new mongoose.Document({}, AcademicProfileTagSchema);
    const endorsement = new mongoose.Document({}, AcademicProfileEndorsementSchema);
    const comment = new mongoose.Document({}, AcademicProfileCommentSchema);

    const doc = new mongoose.Document({
      user: user._id,
      createdAt: new Date(),
      tags: [tag._id],
      endorsements: [endorsement._id],
      comments: [comment._id],
    }, AcademicProfileSchema);

    return doc.validate();
  });
});
