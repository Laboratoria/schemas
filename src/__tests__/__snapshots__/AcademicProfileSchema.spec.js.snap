// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileSchema should fail validation when fields missing 1`] = `
Object {
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
